assignment/assignment.hs

```diff
@@ -10,7 +10,7 @@ import Data.Char(intToDigit)
 -- a positive integer
 
 factorial :: (Eq p, Num p) => p -> p
-factorial n = 0 -- replace me
+factorial n = if n == 1 then 1 else n*factorial(n-1)
 
```

Comment:
This is also a classic place to use pattern matching:

~~~ hs
factorial 1 = 1
factorial n = n * factorial (n - 1)
~~~

This is purely a personal thing, but I've come to prefer pattern matching to conditional logic. To my eye, it's easier to see what is going on.